sherpa/astro/fake.py

```diff
@@ -0,0 +1,179 @@
+#
+#  Copyright (C) 2021 MIT
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+'''Simulate PHA datasets
+
+This module contains the epynoumous function used to simulate
+`~sherpa.astro.data.DataPHA` datasets.
+'''
+import numpy as np
+import sherpa
+from sherpa.utils.err import DataErr
+from sherpa.astro.background import get_response_for_pha
+
+__all__ = ('fake_pha', )
+
+
+def fake_pha(data, model,
+             is_source=True, pileup_model=None,
+             add_bkgs=False, bkg_models={}, id=None):
+    """Simulate a PHA data set from a model.
+
+    This function replaces the counts in a PHA dataset with simulated counts
+    drawn from a model with Poisson noise. For the simulations, all the details
+    already set up in the PHA dataset will be used, including the exposure
+    time, one or more ARFs and RMFs, area and background scalings,
+    grouping, and data quality arrys.
+
+    Including a background component is optional; if requested, the background
+    will be a Poisson draw from the average of all backgrounds that have been
+    set for the input `~sherpa.astro.data.DataPHA`. For each background
```

Comment:
This would need to be either

- :py:class:`~sherpa.astrp.data.DataPHA`
- `sherpa.astro.data.DataPHA`